synapse/lib/types.py

```diff
@@ -304,39 +311,74 @@ def __init__(self, tlib, name, **info):
         CompType.__init__(self, tlib, name, **info)
         self.sepr = info.get('sep',',')
         self.lower = info.get('lower',0)
+        self.recursive = info.get('recursive',0)
```

Comment:
Since the recursion we're using is only "self" recursion, we should be able to detect and set self.recursion=True rather than requiring them to correctly set it in info.